pkg/user/tx_client.go

```diff
@@ -416,6 +418,17 @@ func (client *TxClient) broadcastTx(ctx context.Context, conn *grpc.ClientConn,
 		return nil, err
 	}
 	if resp.TxResponse.Code != abci.CodeTypeOK {
+		if apperrors.IsNonceMismatchCode(resp.TxResponse.Code) {
```

Comment:
do we want to limit retries or retry until we get it right and tx client gets unblocked?

Reply:
I think it's fine. It should also stop if the user cancels the context or hits their deadline